### PR TITLE
Prepared queries for inbox

### PR DESCRIPTION
**Pull request by chrzaszcz**
- Use prepared queries in `mod_inbox_rdbms`
- Make it possible to include update expressions like 'value = column.value + 1' in upsert queries
- Simplify some queries, e.g. property updates

Details in commit messages

### CODE REVIEW

File: src/rdbms/rdbms_queries.erl [hunk 1]
```diff
@@ -89,35 +89,37 @@ execute_upsert(Host, Name, InsertParams, UpdateParams, UniqueKeyValues) ->
 
 %% @doc
 %% This functions prepares query for inserting a row or updating it if already exists
+%% Updates can be either fields or literal expressions like "column = tab.column + 1"
 -spec prepare_upsert(Host :: mongoose_rdbms:server(),
                      QueryName :: atom(),
                      TableName :: atom(),
                      InsertFields :: [binary()],
-                     UpdateFields :: [binary()],
```
Comment (arcusfelis) [inline src/rdbms/rdbms_queries.erl:96]: could be an extra argument `UpdateSQL`, which is empty by default.
And so we would have prepare_upsert/6 and prepare_upsert/7.

But  not thaaat critical, it's called once per statement anyway.

**Review by arcusfelis — APPROVED** [review]
looks good to me

File: src/rdbms/rdbms_queries.erl (hunk 1 above)
Comment (chrzaszcz) [reply src/rdbms/rdbms_queries.erl:96]: I admit I avoided adding the extra argument. I could do it like you suggest to make it more obvious for the user.

**Review by leszke — APPROVED** [review]
Looks good!

**Pull request merged**